.github/workflows/release.yml

```diff
@@ -13,14 +17,48 @@ jobs:
         with:
           fetch-depth: 0
           token: ${{ secrets.RELEASE_GH_TOKEN }}
+
       - uses: actions/setup-node@v3
         with:
           node-version: 18
           cache: 'npm'
+
       - name: Install dependencies and build TS
         run: npm ci && npm run build
+
       - name: Install semantic-release and friends
         run: npm i --no-save semantic-release@20 @semantic-release/changelog@6 @semantic-release/exec@6 @semantic-release/git@10
+
+      # We do a dry run here to analyze the commits and grab the next version
+      # for usage in the Docker metadata action
+      - name: Dry run of semantic-release workflow
+        run: npx semantic-release --dry-run
+        id: release
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          # For every release we add 2-3 tags to the docker build:
+          # 1. A semver tag (e.g., 8.0.0, 8.0.0-next.0)
+          # 2. A branch tag (e.g., next, main)
+          # 3. A `latest` tag (only on the main branch)
+          tags: |
+            type=semver,pattern={{version}},value=${{ steps.release.outputs.nextVersion }}
+            type=ref,event=branch
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
```

Comment:
What's this doing? Setting `latest` to `main` or to the commit sha?

Reply:
say we're on the `main` branch and this workflow is tagging the `8.6.0` release, it'll add the following three tags to the current docker build:
1. `8.6.0`
2. `main` (it'll be `next` if on the `next` branch)
3. `latest` (this will only be added on the `main` branch, which is what that line is doing)

i talk about this here but let me know if I should fluff this out at all:
https://github.com/readmeio/rdme/blob/3b153c84159182499e7dd66fcc276cfcb05e1bfd/.github/workflows/release.yml#L53-L56